hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/api/records/timelineservice/ApplicationEntity.java

```diff
@@ -43,7 +43,12 @@ public ApplicationEntity(TimelineEntity entity) {
   }
 
   public String getQueue() {
-    return getInfo().get(QUEUE_INFO_KEY).toString();
+    if (getInfo() != null) {
```

Comment:
Is this an unrelated change?

Reply:
This is a necessary change, and I have provided a detailed explanation of the modification in the comment below.

Reply:
Thank you for the explanation.

Is this potentially a backward-incompatible change? For example, does it mean clients may now be getting back a queue element with an empty string, where previously it was omitted entirely from the response?

Reply:
Thank you very much for your comment! I have further refined this part of the code and added the `@JsonInclude(JsonInclude.Include.NON_EMPTY)` annotation to ensure compatibility of this logic.

Reply:
From my perspective, we need to continue iterating and improving the YARN Timeline-related code, implementing parsing for both JSON and XML formats in the way Jersey2 does. I hope that after completing the JDK17 support, we can continue advancing these improvements.